### PR TITLE
Create combined Mastodon bot

### PR DESCRIPTION
**Pull request by SolarLiner**
I've combined the codes from all 3 bots into a master bot that launches all 3 at once.

I can't test the bot as I need the API keys and it's too late for me to go into a full debugging session right now. If needed I'll get myself one tomorrow.

### CODE REVIEW

File: mastodon_bot_combined/src/bot2.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      console.log(data.content);
+    }
+  });
+}
+
+module.exports = function bot2() {
+  const listener = M.stream("streaming/user");
+  listener.on("error", err => console.log(err));
+
+  listener.on("message", msg => {
```
Comment (shiffman) [inline mastodon_bot_combined/src/bot2.js]: We probably need to reconcile this with the treegen replies. There are four things that can happen:

1. favorite
2. reblog
3. question
4. generate a tree

If none of these are in a mention, maybe the bot says something more than missing angle?

File: mastodon_bot_combined/src/bot3.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
+const M = require("./mastodon");
+require("dotenv").config();
```
Comment (shiffman) [inline mastodon_bot_combined/src/bot3.js]: Should all of these repeated requirements be a part of `require("./mastodon");`?

File: mastodon_bot_combined/src/bot2.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
+const M = require("./mastodon");
+
+function toot(content, id) {
+  const params = {
+    status: content
+  }
+  if (id) {
+    params.in_reply_to_id = id;
+  }
+  M.post('statuses', params, (error, data) => {
```
Comment (shiffman) [inline mastodon_bot_combined/src/bot2.js]: We might as well refactor everything to use `async` functions rather than callbacks?

**Review by shiffman — COMMENTED** [review]
Thank! A few comments here! And also would love to mix and match with the ideas from #8?

File: mastodon_bot_combined/src/bot2.js (hunk 3 above)
Comment (SolarLiner) [reply mastodon_bot_combined/src/bot2.js]: I've just copy/pasted code from the individual bots to post the PR. I can indeed rework the code into Async/Await code.

**Comment by SolarLiner** [comment]
Also, I don't really know `dotenv` so I don't know whether the `.env` file placement is right or not.

**Pull request merged**